src/services/completions.ts

```diff
@@ -2148,6 +2171,26 @@ namespace ts.Completions {
         return isConstructorParameterCompletionKeyword(stringToToken(text));
     }
 
+    function isFunctionLikeBodyCompletionKeyword(kind: SyntaxKind) {
+        switch (kind) {
+            case SyntaxKind.PublicKeyword:
+            case SyntaxKind.PrivateKeyword:
+            case SyntaxKind.ProtectedKeyword:
+            case SyntaxKind.ReadonlyKeyword:
+            case SyntaxKind.ConstructorKeyword:
+            case SyntaxKind.StaticKeyword:
+            case SyntaxKind.AbstractKeyword:
+            case SyntaxKind.GetKeyword:
+            case SyntaxKind.SetKeyword:
+                return false;
+        }
+        return true;
+    }
+
+    function isFunctionLikeBodyCompletionKeywordText(text: string) {
+        return isFunctionLikeBodyCompletionKeyword(stringToToken(text));

```

Comment:
is not that just `!isClassMemberCompletionKeywordText(text)` ?

Reply:
'async' keyword is not included

Reply:
and in my opinion they should be orthogonal